### PR TITLE
Add support for hook_entity_field_access

### PR DESCRIPTION
**Pull request by barthje**
https://www.drupal.org/node/2919024

### CODE REVIEW

**Review by pdenooijer — CHANGES_REQUESTED** [review]
Please add a PHPUnit test to test your event 🙂

File: src/Event/EntityField/EntityFieldAccessEvent.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+ * @package Drupal\hook_event_dispatcher\Event\EntityField
+ */
+class EntityFieldAccessEvent extends Event implements EventInterface {
+
+  /**
+   * The operation.
+   *
+   * @var string
+   */
+  protected $operation;
```
Comment (pdenooijer) [inline src/Event/EntityField/EntityFieldAccessEvent.php:26]: Pretty sure all variables can be private.

File: src/Event/EntityField/EntityFieldAccessEvent.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  /**
+   * Get the field definition.
+   *
+   * @return \Drupal\Core\Field\FieldDefinitionInterface
+   *   The field definition.
+   */
+  public function getFieldDefinition() {
+    return $this->fieldDefintion;
+  }
+
```
Comment (pdenooijer) [inline src/Event/EntityField/EntityFieldAccessEvent.php:112]: Please add getters for operation, account and items as well.

**Review by pdenooijer — CHANGES_REQUESTED** [review]

**Comment by barthje** [comment]
@pdenooijer just a question, why aren't you using the Prophet class ? Core is using the same thing. I get errors when using createMock. Is it alright if I use prophet?

**Comment by pdenooijer** [comment]
If it's a normal PHPUnit TestCase and it does not add any dependencies its fine by me 👍. Never used Prophet, but I guess it's kinda the same as Mockery/PHPUnit Mock. Then I learn something new 😉!

**Pull request merged**